src/object/cli_obj.c

```diff
@@ -688,24 +689,127 @@ obj_rw_req_reassemb(struct dc_object *obj, daos_obj_rw_t *args,
 	return rc;
 }
 
+static bool
+obj_ec_tgt_in_err(uint32_t *err_list, uint32_t nerrs, uint16_t tgt_idx)
+{
+	uint32_t	i;
+
+	for (i = 0; i < nerrs; i++) {
+		if (err_list[i] == tgt_idx)
+			return true;
+	}
+	return false;
+}
+
+static int
+obj_ec_get_degrade(struct obj_auxi_args *obj_auxi, uint16_t fail_tgt_idx,
+		   uint32_t *parity_tgt_idx)
+{
+	struct obj_reasb_req	*reasb_req = &obj_auxi->reasb_req;
+	uint16_t		 p = obj_ec_parity_tgt_nr(reasb_req->orr_oca);
+	uint16_t		 k = obj_ec_data_tgt_nr(reasb_req->orr_oca);
+	uint32_t		*err_list;
+	uint32_t		 nerrs, i;
+	bool			 with_parity = false;
+
+	D_ASSERT(fail_tgt_idx < k + p);
+	if (reasb_req->orr_err_list == NULL) {
+		D_ALLOC_ARRAY(reasb_req->orr_err_list, p);
+		if (reasb_req->orr_err_list == NULL)
+			return -DER_NOMEM;
+		reasb_req->orr_nerrs = 0;
+	}
+	err_list = reasb_req->orr_err_list;
+	nerrs = reasb_req->orr_nerrs;
+	D_ASSERT(nerrs <= p);
+	if (nerrs == p) {
+		D_ERROR("already with %d error targets, not recoverable.\n", p);
+		return -DER_DATA_LOSS;
+	}
+
+	for (i = 0; i < nerrs; i++)
+		D_ASSERT(err_list[i] != fail_tgt_idx);
+
+	err_list[nerrs] = fail_tgt_idx;
+	reasb_req->orr_nerrs++;
+	nerrs = reasb_req->orr_nerrs;
+
+	for (i = k; i < k + p; i++) {
+		if (!obj_ec_tgt_in_err(err_list, nerrs, i)) {
+			*parity_tgt_idx = i;
+			with_parity = true;
```

Comment:
break?

Reply:
I think you are right, better to break it here. I'll do some test ...